### PR TITLE
Surface driven bubble problem 

### PR DESCRIPTION
**Pull request by akshaysridhar**
RTB with prescribed bottom boundary fluxes. 
1) Adds `LES` `experiment` file. 
2) Exports `vertical_unit_vector`
3) Cleans up dycoms + surfacebubble redundant b.c. function

### CODE REVIEW

File: experiments/LES/surfacebubble.jl [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+  @inbounds begin
+    # Momentum b.c. prescribed (no flow across wall boundary)
+    if typeof(nf) <: NumericalFluxNonDiffusive
+      Y⁺.ρu -= 2 * dot(Y⁻.ρu, n⁻) * SVector(n⁻)
+    else
+      Y⁺.ρu -=  dot(Y⁻.ρu, n⁻) * SVector(n⁻)
+    end
+  end
+  nothing
+end
```
Comment (jkozdon) [inline experiments/LES/surfacebubble.jl]: I'd suggest something like:
```julia
atmos_boundary_state!(nf::Union{NumericalFluxNonDiffusive, NumericalFluxGradient},
                      bc:: SurfaceDrivenBubbleBC, 
                      args...) = atmos_boundary_state!(nf, NoFluxBC(), args...)
```
Similarly to how is done in [dycoms](https://github.com/climate-machine/CLIMA/blob/c688b17762e3ef60233a46f6ba3fe094aa3bc2f6/experiments/LES/dycoms.jl#L52-L61)

File: experiments/LES/surfacebubble.jl [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+                                        bctype, t, _...)
+  FT = eltype(Y⁻)
+  @inbounds begin
+    # Momentum b.c. prescribed (no flow across wall boundary)
+    if typeof(nf) <: NumericalFluxNonDiffusive
+      Y⁺.ρu -= 2 * dot(Y⁻.ρu, n⁻) * SVector(n⁻)
+    else
+      Y⁺.ρu -=  dot(Y⁻.ρu, n⁻) * SVector(n⁻)
+    end
+  end
```
Comment (jkozdon) [inline experiments/LES/surfacebubble.jl]:
```julia
    atmos_boundary_state!(nf, NoFluxBC(), m,
                          Y⁺, Σ⁺, A⁺, n⁻,
                          Y⁻, Σ⁻, A⁻,
                          bctype, t)
```
similar to [dycoms](https://github.com/climate-machine/CLIMA/blob/c688b17762e3ef60233a46f6ba3fe094aa3bc2f6/experiments/LES/dycoms.jl#L89-L92)

File: experiments/LES/surfacebubble.jl [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+                         Y⁺, Σ⁺, A⁺, n⁻,
+                         Y⁻, Σ⁻, A⁻,
+                         bctype, t,
+                         Y₁⁻, Σ₁⁻, A₁⁻) =
+  atmos_boundary_flux_diffusive!(nf, atmos.boundarycondition, atmos,
+                                 F,
+                                 Y⁺, Σ⁺, A⁺, n⁻,
+                                 Y⁻, Σ⁻, A⁻,
+                                 bctype, t,
+                                 Y₁⁻, Σ₁⁻, A₁⁻)
```
Comment (jkozdon) [inline experiments/LES/surfacebubble.jl]: Why is this needed in the driver? Seems to be duplicated code from the [AtmosModel](https://github.com/climate-machine/CLIMA/blob/c688b17762e3ef60233a46f6ba3fe094aa3bc2f6/src/Atmos/Model/AtmosModel.jl#L320-L339).

The same question applies to [dycoms](https://github.com/climate-machine/CLIMA/blob/c688b17762e3ef60233a46f6ba3fe094aa3bc2f6/experiments/LES/dycoms.jl#L142-L156)

File: experiments/LES/surfacebubble.jl [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+    if typeof(nf) <: NumericalFluxNonDiffusive
+      Y⁺.ρu -= 2 * dot(Y⁻.ρu, n⁻) * SVector(n⁻)
+    else
+      Y⁺.ρu -=  dot(Y⁻.ρu, n⁻) * SVector(n⁻)
+    end
+  end
+
+  k̂  = vertical_unit_vector(m.orientation, A⁻)
+  r = sqrt((A⁻.coord[1]-bc.x₀)^2 + (A⁻.coord[2]-bc.x₀)^2) 
+  F₀ =  bc.F₀*(1 - sign(t-bc.t₁))/2
```
Comment (jkozdon) [inline experiments/LES/surfacebubble.jl]:
```suggestion
  F₀ =  t < bc.t₁ ? bc.F₀ : -zero(FT)
```

File: experiments/LES/surfacebubble.jl [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+    # Momentum b.c. prescribed (no flow across wall boundary)
+    if typeof(nf) <: NumericalFluxNonDiffusive
+      Y⁺.ρu -= 2 * dot(Y⁻.ρu, n⁻) * SVector(n⁻)
+    else
+      Y⁺.ρu -=  dot(Y⁻.ρu, n⁻) * SVector(n⁻)
+    end
+  end
+
+  k̂  = vertical_unit_vector(m.orientation, A⁻)
+  r = sqrt((A⁻.coord[1]-bc.x₀)^2 + (A⁻.coord[2]-bc.x₀)^2)
```
Comment (jkozdon) [inline experiments/LES/surfacebubble.jl]:
```suggestion
  r = hypot(A⁻.coord[1] - bc.x₀, A⁻.coord[2] - bc.x₀) 
```

File: experiments/LES/surfacebubble.jl (hunk 1 above)
Comment (akshaysridhar) [reply experiments/LES/surfacebubble.jl]: Noted, thanks !

File: experiments/LES/surfacebubble.jl (hunk 2 above)
Comment (akshaysridhar) [reply experiments/LES/surfacebubble.jl]: Noted - will use the `NoFluxBC` default

File: experiments/LES/surfacebubble.jl (hunk 3 above)
Comment (akshaysridhar) [reply experiments/LES/surfacebubble.jl]: @jkozdon: fixed in both files.

**Comment by akshaysridhar** [comment]
bors try

File: experiments/LES/surfacebubble.jl [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+Σ ≡ diffusive vars
+A ≡ auxiliary vars
+X⁺ and X⁻ refer to exterior, interior faces
+X₁ refers to the first interior node 
+
+# Fields
+$(DocStringExtensions.FIELDS)
+"""
+struct SurfaceDrivenBubbleBC{FT} <: BoundaryCondition
+  "Prescribed MSEF Magnitude [W/m^2]"
```
Comment (charleskawczynski) [inline experiments/LES/surfacebubble.jl]:
```suggestion
  "Prescribed MSEF Magnitude `[W/m^2]`"
```

Otherwise the doc strings break (due to looking for a link)

Comment (akshaysridhar) [reply experiments/LES/surfacebubble.jl]: Thanks - will sort in other instances where this occurs as well.

**Comment by akshaysridhar** [comment]
bors try

**Comment by akshaysridhar** [comment]
> I made a comment on docstrings containing brackets-- documenter will try to find a link and break, so we should include them in backticks, or just remove the brackets.
> 
> Other than that, looks good! I'm happy to see that a new driver is only ~200 lines!

Your suggestion's been included 👍

**Comment by akshaysridhar** [comment]
bors try
